citadel_workspace_service/tests/file_transfer.rs

```diff
@@ -0,0 +1,898 @@
+                }*/
+            } else {
+                panic!("File Transfer P2P Failure");
+            }
+        } else {
+            panic!("File Transfer Request failed: {deserialized_service_a_payload_response:?}");
+        }
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_citadel_workspace_service_c2s_revfs() -> Result<(), Box<dyn Error>> {
+        citadel_logging::setup_log();
+        info!(target: "citadel", "above server spawn");
+        let bind_address_internal_service: SocketAddr = "127.0.0.1:55518".parse().unwrap();
+
+        // TCP client (GUI, CLI) -> Internal Service -> Receiver File Transfer Kernel server
+        let server_success = &Arc::new(AtomicBool::new(false));
+        let (server, server_bind_address) = server_info_file_transfer(server_success.clone());
+
+        tokio::task::spawn(server);
+
+        info!(target: "citadel", "sub server spawn");
+        let internal_service_kernel = CitadelWorkspaceService::new(bind_address_internal_service);
+        let internal_service = NodeBuilder::default()
+            .with_node_type(NodeType::Peer)
+            .with_backend(BackendType::InMemory)
+            .build(internal_service_kernel)?;
+
+        tokio::task::spawn(internal_service);
+
+        // give time for both the server and internal service to run
+
+        tokio::time::sleep(Duration::from_millis(2000)).await;
+
+        info!(target: "citadel", "about to connect to internal service");
+
+        let (to_service, mut from_service, uuid, cid) = register_and_connect_to_server(
+            bind_address_internal_service,
+            server_bind_address,
+            "John Doe",
+            "john.doe",
+            "secret",
+        )
+        .await
+        .unwrap();
+
+        // Push file to REVFS
+        let file_to_send = PathBuf::from("../../resources/test.txt");
+        let virtual_path = PathBuf::from("vfs/");
+        let file_transfer_command = InternalServicePayload::SendFile {
+            uuid,
+            source: file_to_send,
+            cid,
+            is_refvs: true,
+            peer_cid: None,
+            chunk_size: None,
+            virtual_directory: Some(virtual_path),
+            security_level: None,
+        };
+        to_service.send(file_transfer_command).unwrap();
+        let file_transfer_response = from_service.recv().await.unwrap();
+        match file_transfer_response {
+            InternalServiceResponse::SendFileSuccess(SendFileSuccess { cid: response_cid }) => {
+                assert_eq!(cid, response_cid);
+            }
+            _ => {
+                info!(target = "citadel", "{:?}", file_transfer_response);
+                panic!("Didn't get the REVFS SendFileSuccess");
+            }
+        }
+        info!(target: "citadel","{file_transfer_response:?}");
+
+        // Download/Pull file from REVFS - Don't delete on pull
+        let virtual_path = PathBuf::from("vfs/virtual_test.txt");
```

Comment:
Should be consistent with above virtual path (might as well just re-use the variable from above)
```suggestion
        let virtual_path = PathBuf::from("/test.txt");
```

OR

```suggestion
        let virtual_path = virtual_path.clone()
```